### PR TITLE
[e2e] Should spread Pods to schedulable cluster zones

### PR DESCRIPTION
**Pull request by lzhecheng**
Some Nodes like control plane ones should not be considered
to spread Pods.

related: https://github.com/kubernetes-sigs/cloud-provider-azure/issues/919

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Some Nodes like control plane ones should not be considered
to spread Pods.
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/scheduling/ubernetes_lite.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string)
 	framework.ExpectNoError(err)
 
 	// Now make sure they're spread across zones
-	zoneNames, err := e2enode.GetClusterZones(f.ClientSet)
+	zoneNames, err := e2enode.GetSchedulableClusterZones(f.ClientSet)
```
Comment (kerthcet) [inline test/e2e/scheduling/ubernetes_lite.go:133]: IIUC, before https://github.com/kubernetes/enhancements/issues/3094, `PodTopologySpread` plugin will ignore taints in calculation, which means master nodes are also taken into account. So `GetClusterZones` helps to validate this logic.

Comment (lzhecheng) [reply test/e2e/scheduling/ubernetes_lite.go:133]: Is it going to allow Pods to be allocated to control plane nodes?

Comment (kerthcet) [reply test/e2e/scheduling/ubernetes_lite.go:133]: Scheduling decisions are made by the Filter/Score results, not the role of the nodes. If control plane nodes have the untolerated taint like `node-role.kubernetes.io/master:NoSchedule`, they're still unschedulable. I have no idea whether the work nodes also have untolerated taints etc. but filtered out by `GetSchedulableClusterZones` which escaped from our validation in `checkZoneSpreading`.
https://github.com/kubernetes/kubernetes/blob/226323178e23b4c476001266beab1e2f116b3879/test/e2e/framework/node/resource.go#L600

Comment (lzhecheng) [reply test/e2e/scheduling/ubernetes_lite.go:133]: control plane nodes have untolerated taint and pods won't be allocated there. `GetClusterZones` returns all zones including ones control plane nodes are on. As a result, these control plane zones get no Pod and validation fails. Is my understanding correct?

Comment (kerthcet) [reply test/e2e/scheduling/ubernetes_lite.go:133]: Ah, I misunderstood the intension of the tests. Narrow down the available zones make sense to me. Sorry for that.

File: test/e2e/scheduling/ubernetes_lite.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func getZoneNameForNode(node v1.Node) (string, error) {
 
 // Return the number of zones in which we have nodes in this cluster.
 func getZoneCount(c clientset.Interface) (int, error) {
-	zoneNames, err := e2enode.GetClusterZones(c)
+	zoneNames, err := e2enode.GetSchedulableClusterZones(c)
```
Comment (kerthcet) [inline test/e2e/scheduling/ubernetes_lite.go:151]: From the comment, it seems like we should include all nodes.

Comment (lzhecheng) [reply test/e2e/scheduling/ubernetes_lite.go]: Since we reach the agreement, I think all zones should be schedulable ones. Updated.

**Comment by lzhecheng** [comment]
@damemi @sanposhiho could you please take a look? Thanks.

**Review by damemi — COMMENTED** [review]
/approve

**Pull request merged**